pkg/dotnet/core_deps/parse.go

```diff
@@ -31,40 +47,60 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 	}
 
 	var libraries []types.Library
-	for nameVer, lib := range depsFile.Libraries {
-		if !strings.EqualFold(lib.Type, "package") {
+	var deps []types.Dependency
+	targets := depsFile.Targets[depsFile.RuntimeTarget.Name]
+	for pkgNameVersion, target := range targets {
+		name, version := splitNameVer(pkgNameVersion)
+		if name == "" || version == "" {
 			continue
 		}
 
-		split := strings.Split(nameVer, "/")
-		if len(split) != 2 {
-			// Invalid name
-			log.Logger.Warnf("Cannot parse .NET library version from: %s", nameVer)
-			continue
+		lib := types.Library{
+			ID:        packageID(name, version),
+			Name:      name,
+			Version:   version,
+			Locations: []types.Location{{StartLine: target.StartLine, EndLine: target.EndLine}},
+		}
+
+		var childDeps []string
+		for depName, depVersion := range target.Dependencies {
+			depID := packageID(depName, depVersion)
+			if _, ok := targets[depID]; ok {
+				childDeps = append(childDeps, depID)
+			}
 		}
 
-		libraries = append(libraries, types.Library{
-			Name:      split[0],
-			Version:   split[1],
-			Locations: []types.Location{{StartLine: lib.StartLine, EndLine: lib.EndLine}},
-		})
+		if len(childDeps) > 0 {
+			deps = append(deps, types.Dependency{
+				ID:        lib.ID,
+				DependsOn: childDeps,
+			})
+		}
+
+		libraries = append(libraries, lib)
 	}
 
-	return libraries, nil, nil
+	return libraries, deps, nil
```

Comment:
`Libraries` and `Dependencies` can be sorted.
```suggestion
	sort.Sort(types.Libraries(libs))
	sort.Sort(types.Dependencies(deps))
	
	return libraries, deps, nil
```

After these changes you can remove sorting from tests.